packages/application-shell/src/hooks/use-all-menu-feature-toggles/use-all-menu-feature-toggles.spec.js

```diff
@@ -0,0 +1,105 @@
+import React from 'react';
+import { reportErrorToSentry } from '@commercetools-frontend/sentry';
+import { renderApp, waitForElement, wait } from '../../test-utils';
+import FetchAllMenuFeatureToggles from './fetch-all-menu-feature-toggles.proxy.graphql';
+import useAllMenuFeatureToggles from './use-all-menu-feature-toggles';
+
+jest.mock('@commercetools-frontend/sentry');
+
+const TestComponent = () => {
+  const { isLoading, allFeatureToggles } = useAllMenuFeatureToggles();
+
+  if (!isLoading) {
+    return (
+      <>
+        <h3>Number of toggles: {allFeatureToggles.length}</h3>
+        <ul>
+          {allFeatureToggles.map(featureToggle => (
+            <div key={featureToggle}>{`Toggle: ${featureToggle}`}</div>
+          ))}
+        </ul>
+      </>
+    );
+  }
+
+  return <div>{'Loading'}</div>;
+};
+
+const mockRequests = {
+  withoutError: {
+    request: {
+      query: FetchAllMenuFeatureToggles,
+    },
+    result: {
+      data: {
+        allFeatureToggles: ['flagA', 'flagB'],
+      },
+    },
+  },
+  withError: {
+    request: {
+      query: FetchAllMenuFeatureToggles,
+    },
+    result: {
+      errors: [
+        {
+          message: 'There has been an error.',
+        },
+      ],
+    },
+  },
+};
+
+const render = ({ mocks, environment } = { mocks: [] }) =>
+  renderApp(<TestComponent />, {
+    mocks,
+    environment,
+  });
+
+describe('when served by proxy', () => {
+  describe('without error', () => {
+    it('should be loading and return any feature toggles', async () => {
+      const rendered = render({
+        environment: {
+          servedByProxy: true,
+        },
+        mocks: [mockRequests.withoutError],
+      });
+
+      await waitForElement(() => rendered.getByText('Loading'));
+      await waitForElement(() => [
+        rendered.getByText(/Number of toggles: 2/i),
+        rendered.getByText(/Toggle: flagA/i),
+        rendered.getByText(/Toggle: flagB/i),
+      ]);
+    });
+  });
+
+  describe('with error', () => {
```

Comment:
I think we can remove this test, it’s not really super important. It has also been kind of flaky in the past, not sure if with the more recent versions of apollo it got more stable or not.

Reply:
Hm, thanks. I'll keep it and see how stable it'll be. The ones in the MC for signup etc include loads of error states and are quite stable.